### DISCUSSION TITLE
MD3 Date Picker

### DISCUSSION THREAD

**Discussion by iM-GeeKy** [web-ridge/react-native-paper-dates#224] [General] [answered]
### Motivation
Currently our date picker for Material Design 3 does not meet the [documentation specifications](https://m3.material.io/components/date-pickers/specs). Furthermore, it appears there are different pickers that aren't all full screen height.
![image](https://user-images.githubusercontent.com/7604441/209669299-4f300293-d3c5-45ac-8065-fb473122150b.png)

@RichardLindhout Would you be interested in tackling this? It seems more involved and might cause us to have separate pickers for between version 2 and 3. I'd be glad to discuss a plan of attack on this.

**Comment by RichardLindhout** [comment, marked as answer]
I think the docked ones are (only on big screens) also in m2 but were never implemented
https://m2.material.io/components/date-pickers#desktop-pickers
Open issue: https://github.com/web-ridge/react-native-paper-dates/issues/9

They are also hard because they need take into account the scrolling position of the parent (which can be really hard)
I think we can best implement the modal pickers first in m3 (with the same components)
https://m3.material.io/components/date-pickers/specs


The only way to work with the docked ones is to know the scrollview scroll position which is a hard problem (e.g. Menu component of RNP does not work great on web, because you can't scroll if menu is open)

We could take inspiration of this library, but this still needs some love and fix some bugs, but maybe I could extract some basic components like the ScrollView + useOffsetX, offsetY. But it would probably add a dependency on react-native-reanimated.
https://github.com/web-ridge/react-native-paper-autocomplete

**Reply by iM-GeeKy** [reply]
It might make sense to just iterate on it rather than trying to fix it all on the first go. That would start with just updating the date pickers such that it uses the proper color scheme, text, etc for a more consistent feel.

**Reply by iM-GeeKy** [reply]
I think I can get around to starting this in a couple of days. I'll plan to just update the colors schemes, etc, and we can punt the other upgrades for a later request.
